test/local/devices_tests.js

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var uuid = require('uuid')
+var crypto = require('crypto')
+var test = require('../ptaptest')
+var mocks = require('../mocks')
+
+var modulePath = '../../lib/devices'
+
+test('require', function (t) {
+  t.equal(typeof require(modulePath), 'function', 'require returns function')
+  t.equal(require(modulePath).length, 3, 'returned function expects three arguments')
+
+  test('instantiate', function (t) {
+    var log = mocks.spyLog()
+    var deviceCreatedAt = Date.now()
+    var deviceId = crypto.randomBytes(16)
+    var device = {
+      name: 'foo',
+      type: 'bar'
+    }
+    var db = mocks.mockDB({
+      device: device,
+      deviceCreatedAt: deviceCreatedAt,
+      deviceId: deviceId
+    })
+    var push = mocks.mockPush()
+    var devices = require(modulePath)(log, db, push)
+
+    t.equal(typeof devices, 'object', 'devices is object')
+    t.equal(Object.keys(devices).length, 2, 'devices has two properties')
+
+    t.equal(typeof devices.upsert, 'function', 'devices has upsert method')
+    t.equal(devices.upsert.length, 3, 'devices.upsert expects three arguments')
+
+    t.equal(typeof devices.synthesizeName, 'function', 'devices has synthesizeName method')
+    t.equal(devices.synthesizeName.length, 1, 'devices.synthesizeName expects 1 argument')
```

Comment:
Ha, quite the opposite actually, I much prefer dynamic typing.

However I also think that, in lieu of static types, it pays to have an abundance of assertions because programs tend to break in ways you don't expect.